src/Form.php

```diff
@@ -0,0 +1,378 @@
+     *         'Email address is invalid.',
+     *     ]
+     * ]
+     * ```
+     *
+     * {@see getFirstErrors()}
+     * {@see getFirstError()}
+     */
+    public function getErrors(?string $attribute = null): array
+    {
+        if ($attribute === null) {
+            return $this->errors === null ? [] : $this->errors;
+        }
+
+        return isset($this->errors[$attribute]) ? $this->errors[$attribute] : [];
+    }
+
+    /**
+     * Returns the first error of the specified attribute.
+     *
+     * @param string $attribute attribute name.
+     *
+     * @return string|null the error message. Null is returned if no error.
+     *
+     * {@see getErrors()}
+     * {@see getFirstErrors()}
+     */
+    public function getFirstError(string $attribute): string
+    {
+        return isset($this->errors[$attribute]) ? reset($this->errors[$attribute]) : '';
+    }
+
+    /**
+     * Returns the first error of every attribute in the model.
+     *
+     * @return array the first errors. The array keys are the attribute names, and the array values are the
+     * corresponding error messages. An empty array will be returned if there is no error.
+     *
+     * {@see getErrors()}
+     * {@see getFirstError()}
+     */
+    public function getFirstErrors(): array
+    {
+        if (empty($this->errors)) {
+            return [];
+        }
+
+        $errors = [];
+
+        foreach ($this->errors as $name => $es) {
+            if (!empty($es)) {
+                $errors[$name] = \reset($es);
+            }
+        }
+
+        return $errors;
+    }
+
+    /**
+     * Returns a value indicating whether there is any validation error.
+     *
+     * @param string|null $attribute attribute name. Use null to check all attributes.
+     *
+     * @return bool whether there is any error.
+     */
+    public function hasErrors(?string $attribute = null): bool
+    {
+        return $attribute === null ? !empty($this->errors) : isset($this->errors[$attribute]);
+    }
+
+    /**
+     * Populates the model with input data.
+     *
+     * which, with `load()` can be written as:
+     *
+     * ```php
+     * $body = $request->getParsedBody();
+     * $method = $request->getMethod();
+     *
+     * if ($method === Method::POST && $loginForm->load($body)) {
+     *     // handle success
+     * }
+     * ```
+     *
+     * `load()` gets the `'FormName'` from the model's [[formName()]] method (which you may override), unless the
+     * `$formName` parameter is given. If the form name is empty, `load()` populates the model with the whole of
+     * `$data`,
+     * instead of `$data['FormName']`.
+     *
+     * Note, that the data being populated is subject to the safety check by [[setAttributes()]].
+     *
+     * @param array $data the data array to load, typically `$_POST` or `$_GET`.
+     *
+     * @return bool whether `load()` found the expected form in `$data`.
+     */
+    public function load($data): bool
```

Comment:
How about to add this method to the interface?

Reply:
done.